### PR TITLE
refactor/LW-7148 - PortfolioBalance refactor

### PR DESCRIPTION
**Pull request by lgobbi-atix**
# Checklist

- [x] JIRA - LW-7148
- [x] Proper tests implemented

---

## Proposed solution

Continuation of: https://github.com/input-output-hk/lace/pull/170

Refactor `PortfolioBalance.tsx` component to make it easier to understand and modify

- Separate into smaller components
- Rename variables and remove unnecessary ones
- Improve unit testing

## Testing

Portfolio balance in Tokens screen, balances in asset details and show/hide balance toggle should work the same way as before in both browser and popup views:

image
image

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/1067/5599044782/index.html) for [a95d97a9](https://github.com/input-output-hk/lace/pull/186/commits/a95d97a97f14b5ae610fc4b4bc92f6ac9f134e15)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 35     | 0      | 0       | 0     | 35    | ✅     |

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    color: var(--text-color-secondary);
+    font-weight: 700;
+    text-align: left;
+    margin: 0;
+  }
+
+  .codeTextLarge {
+    font-size: var(--subHeading);
+    line-height: size_unit(4);
+    font-weight: 700;
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss:21]:
```suggestion
    @includes text-subHeading-bold;
```

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+.neutral {
+  color: var(--text-color-secondary) !important;
+}
+
+.popupView {
+  &.balance {
+    .balanceTextLarge {
+      font-size: var(--pageHeading);
+      line-height: size_unit(7);
+      font-weight: 700;
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss:87]:
```suggestion
     @includes text-pageHeading;
```

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    .balanceTextXLarge {
+      font-size: var(--display);
+      line-height: size_unit(8);
+      font-weight: 700;
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss:92]:
```suggestion
      @includes text-display;
```

**Review by vetalcore — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss (hunk 3 above)
Comment (lgobbi-atix) [reply apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss]: Done a60294309a47e1e8decc88ac2d9ae02171e57873

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss (hunk 2 above)
Comment (lgobbi-atix) [reply apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss]: Done a60294309a47e1e8decc88ac2d9ae02171e57873

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss (hunk 1 above)
Comment (lgobbi-atix) [reply apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss]: Done a60294309a47e1e8decc88ac2d9ae02171e57873

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceLabel/PortfolioBalanceLabel.module.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  display: flex;
+  gap: size_unit(0.5);
+  margin-top: size_unit(1.5);
+  span {
+    display: flex;
+  }
+
+  label {
+    color: $theme-text-color-light-grey;
+    font-size: size_unit(2);
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceLabel/PortfolioBalanceLabel.module.scss:14]: `@include text-bodyLarge-semi-bold`

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    color: $theme-text-color-black;
+    font-weight: 600;
+    text-align: left;
+    margin: 0;
+  }
+
+  .balanceTextLarge {
+    font-size: var(--display);
+    @media (max-width: $breakpoint-popup) {
+      font-size: var(--pageHeading);
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss:39]: `@include text-pageHeading`

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      line-height: size_unit(7);
+    }
+  }
+
+  .balanceTextMedium {
+    font-size: var(--pageHeading);
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss:45]: `@include text-pageHeading`

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    .codeTextLarge {
+      font-size: var(--bodyLarge);
+      line-height: size_unit(4);
+      font-weight: 700;
+    }
+
+    .codeTextMedium {
+      font-size: var(--body);
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss:102]: `@include text-body-bold`

**Review by mchappell — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss (hunk 7 above)
Comment (lgobbi-atix) [reply apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.module.scss]: Done https://github.com/input-output-hk/lace/pull/186/commits/e3904016ab2fdcf4176fff2e8dc773319ae98e6f

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+    parsedSubtitleValue = formatPercentage(parsedSubtitleValue);
+  }
+
+  return (
+    <>
+      <div className={classnames(styles.balance, { [styles.popupView]: isPopupView })}>
+        <h1
+          data-testid="portfolio-balance-value"
+          className={classnames({
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.tsx:66]: please extract + memoise this like you've already done with `subtitleStyles`

File: apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+            [styles.balanceTextMedium]: textSize === 'medium'
+          })}
+        >
+          {isBalanceVisible ? balance : getHiddenBalancePlaceholder()}
+        </h1>
+        {isBalanceVisible && (
+          <h4
+            data-testid="portfolio-balance-currency"
+            className={classnames({
+              [styles.codeTextLarge]: textSize === 'large',
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/components/PortfolioBalance/PortfolioBalanceValue/PortfolioBalanceValue.tsx:78]: please extract + memoise this like you've already done with `subtitleStyles`

**Review by mchappell — CHANGES_REQUESTED** [review]

**Review by mchappell — APPROVED** [review]

**Review by vetalcore — APPROVED** [review]

**Review by ljagiela — APPROVED** [review]

**Pull request merged**